### PR TITLE
added options ui

### LINKED ISSUE #16: Create Option UI
Create user interface for options. See the wireframe below:
setting ui

### PR DESCRIPTION
**Pull request by meomancer**
This fix #16 

screenshot from 2016-12-19 14 57 30
screenshot from 2016-12-19 14 57 36
screenshot from 2016-12-19 14 57 44

### CODE REVIEW

**Comment by ismailsunni** [comment]
LGTM

File: cadasta/gui/tools/cadasta_login.py [hunk 1]
```diff
@@ -20,44 +20,92 @@
...
+
+    def init_style(self):
+        self.setStyleSheet("background-color:white")
+        self.disable_button(self.save_button)
+        self.enable_button(self.test_connection_button)
+        self.test_connection_button.clicked.connect(self.login)
+        self.save_button.clicked.connect(self.save_authtoken)
+
+    def enable_button(self, custom_button):
+        """Enable button"""
```
Comment (Gustry) [inline cadasta/gui/tools/cadasta_login.py]: You should have docstring for each function, with `:param` and `:type`

File: cadasta/gui/tools/cadasta_login.py [hunk 2]
```diff
@@ -20,44 +20,92 @@
...
         super(CadastaLogin, self).__init__(parent)
         self.setupUi(self)
-        self.login_button.clicked.connect(self.login)
         self.msg_bar = None
+        self.text_test_connection_button = self.test_connection_button.text()
+        self.ok_label.setVisible(False)
+        self.init_style()
+
+    def init_style(self):
+        self.setStyleSheet("background-color:white")
```
Comment (Gustry) [inline cadasta/gui/tools/cadasta_login.py:52]: Missing docstring

File: cadasta/gui/tools/cadasta_login.py [hunk 3]
```diff
@@ -20,44 +20,92 @@
...
+        self.auth_token = None
+
+        self.disable_button(self.save_button)
+        self.ok_label.setVisible(False)
 
-        if not username or not password:
+        if not url or not username or not password:
             self.msg_bar = QgsMessageBar()
-            self.msg_bar.pushWarning("Error", "Username/password is empty.")
+            self.msg_bar.pushWarning("Error", "URL/Username/password is empty.")
```
Comment (Gustry) [inline cadasta/gui/tools/cadasta_login.py]: You should be able to translate messages. Use a `tr()`

File: cadasta/gui/tools/cadasta_login.py [hunk 4]
```diff
@@ -20,44 +20,92 @@
...
 
     def on_finished(self, result):
+        self.ok_label.setVisible(True)
         """On finished function when tools request is finished"""
         if 'auth_token' in result:
-            auth_token = result['auth_token']
-            output_result = "auth_token is %s" % auth_token
+            self.auth_token = result['auth_token']
+            self.enable_button(self.save_button)
+            self.ok_label.setText("Success")
```
Comment (Gustry) [inline cadasta/gui/tools/cadasta_login.py]: please use `tr` everywhere in the code.

File: cadasta/utilities/resources.py [hunk 5]
```diff
@@ -95,3 +95,15 @@ def get_ui_class(ui_file):
         )
     )
     return uic.loadUiType(ui_file_path)[0]
+
+
+def get_project_path():
+    """Get Project Path
```
Comment (Gustry) [inline cadasta/utilities/resources.py]: Can you add the docstring ?

File: cadasta/gui/tools/cadasta_login.py [hunk 6]
```diff
@@ -20,44 +20,92 @@
...
+from cadasta.utilities.resources import get_ui_class, get_project_path
 
 from cadasta.api.login import Login
 
+LOGGER = logging.getLogger('CadastaQGISPlugin')
 FORM_CLASS = get_ui_class('cadasta_login_base.ui')
 
 
 class CadastaLogin(QtGui.QDialog, FORM_CLASS):
+    def button_style(self):
```
Comment (Gustry) [inline cadasta/gui/tools/cadasta_login.py]: Minor thing to fix, we can add the `@staticmethod` decorator.

File: cadasta/gui/tools/cadasta_login.py [hunk 7]
```diff
@@ -20,44 +20,92 @@
...
-        if not username or not password:
+        if not url or not username or not password:
             self.msg_bar = QgsMessageBar()
-            self.msg_bar.pushWarning("Error", "Username/password is empty.")
+            self.msg_bar.pushWarning("Error", "URL/Username/password is empty.")
         else:
-            self.login_button.setEnabled(False)
-            self.output_label.setText("logging in....")
+            self.disable_button(self.test_connection_button)
+            self.test_connection_button.setText("logging in")
```
Comment (Gustry) [inline cadasta/gui/tools/cadasta_login.py]: Maybe we should use a capital letter ?
Also, we use single quote for strings in InaSAFE.

File: cadasta/gui/tools/cadasta_login.py [hunk 8]
```diff
@@ -35,29 +38,92 @@ def __init__(self, parent=None):
...
+        self.test_connection_button.clicked.connect(self.login)
+        self.save_button.clicked.connect(self.save_authtoken)
+
+    def enable_button(self, custom_button):
+        """
+        Enable button
+        :param custom_button: button that is enabled
+        :type custom_button: QWidget
+
+        :return
```
Comment (Gustry) [inline cadasta/gui/tools/cadasta_login.py:64]: If there is no 'return' we don't have to put it

File: cadasta/utilities/resources.py [hunk 9]
```diff
@@ -95,3 +95,17 @@ def get_ui_class(ui_file):
         )
     )
     return uic.loadUiType(ui_file_path)[0]
+
+
+def get_project_path():
+    """
+    Get absolute project path
+    :return absolute project path
```
Comment (Gustry) [inline cadasta/utilities/resources.py:103]: We also add ':rtype: basetring'
PyCharm will be aware that a string is returned from the function.

**Pull request merged**